os/board/common/partitions.c

```diff
@@ -91,15 +91,23 @@ void configure_partitions(void)
 			lldbg("ERROR: failed to create partition.\n");
 			return;
 		}
-#if defined(CONFIG_MTD_FTL)
+#ifdef CONFIG_MTD_FTL
 		if (!strncmp(types, "ftl,", 4)) {
 			if (ftl_initialize(partno, mtd_part)) {
 				lldbg("ERROR: failed to initialise mtd ftl errno :%d\n", errno);
 				return;
 			}
 		} else
+#ifdef CONFIG_FS_ROMFS
+		if (!strncmp(types, "romfs,", 6)) {
```

Comment:
1. How about merge this and above?
```
#ifdef CONFIG_MTD_FTL
  if (!strncmp(types, "ftl,", 4)
#ifdef CONFIG_FS_ROMFS
    || !strncmp(types, "romfs,", 6)
#endif
  }
    if (ftl_initialize(xxx)) {
       ...
     }
  } else
#endif
```
2. I can't match up your change with your title.
3. If ROMFS is available with FTL enabled, why don't you make a dependency ?

Reply:
okay. i will modify